tests/python_client/testcases/test_utility.py

```diff
@@ -2309,3 +2390,1259 @@ def test_role_list_grants(self, host, port):
+
+        # Collection permission deny
+        collection_w.load(check_task=CheckTasks.check_permission_deny)
+        collection_w.release(check_task=CheckTasks.check_permission_deny)
+        collection_w.compact(check_task=CheckTasks.check_permission_deny)
+        data = cf.gen_default_list_data(ct.default_nb)
+        mutation_res, _ = collection_w.insert(data=data, check_task=CheckTasks.check_permission_deny)
+        tmp_expr = f'{ct.default_int64_field_name} in {[0]}'
+        collection_w.delete(tmp_expr, check_task=CheckTasks.check_permission_deny)
+        self.index_wrap.drop(ct.default_int64_field_name, check_task=CheckTasks.check_permission_deny)
+        self.index_wrap.init_index(collection_w.collection, ct.default_int64_field_name,
+                                   default_index_params, check_task=CheckTasks.check_permission_deny)
+        vectors = [[random.random() for _ in range(ct.default_dim)] for _ in range(ct.default_nq)]
+        collection_w.search(vectors[:ct.default_nq], ct.default_float_vec_field_name,
+                            ct.default_search_params, ct.default_limit,
+                            "int64 >= 0", check_task=CheckTasks.check_permission_deny)
+        collection_w.flush(check_task=CheckTasks.check_permission_deny)
+        default_term_expr = f'{ct.default_int64_field_name} in [0, 1]'
+        collection_w.query(default_term_expr, check_task=CheckTasks.check_permission_deny)
+        # self.utility_wrap.bulk_load(c_name, check_task=CheckTasks.check_permission_deny)
+
+        # Global permission deny
+        self.init_collection_wrap(name=c_name_2, check_task=CheckTasks.check_permission_deny)
+        collection_w.drop(check_task=CheckTasks.check_permission_deny)
+        self.utility_wrap.create_user(user=c_name, password=password, check_task=CheckTasks.check_permission_deny)
+        self.utility_wrap.init_role(r_name)
+        self.utility_wrap.create_role(check_task=CheckTasks.check_permission_deny)
+        self.utility_wrap.delete_user(user=user, check_task=CheckTasks.check_permission_deny)
+        self.utility_wrap.role_drop(check_task=CheckTasks.check_permission_deny)
+        self.utility_wrap.list_usernames(check_task=CheckTasks.check_permission_deny)
+        self.utility_wrap.role_list_grants(check_task=CheckTasks.check_permission_deny)
+        self.utility_wrap.list_roles(False, check_task=CheckTasks.check_permission_deny)
+        self.utility_wrap.role_add_user(user, check_task=CheckTasks.check_permission_deny)
+        self.utility_wrap.role_remove_user(user, check_task=CheckTasks.check_permission_deny)
+        self.utility_wrap.role_grant("Collection", c_name, "Insert", check_task=CheckTasks.check_permission_deny)
+        self.utility_wrap.role_revoke("Collection", c_name, "Insert", check_task=CheckTasks.check_permission_deny)
+
+        # User permission deny
+        self.utility_wrap.reset_password(user=user_test, old_password=password_test, new_password=password,
```

Comment:
It seems the interface "update_password" has not been covered.

Reply:
in test case test_verify_user_update_privilege